# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../kvm/14-maccpuid.patch
# Copyright (C) 2009 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

Index: kvm-83/qemu/target-i386/helper.c
===================================================================
--- kvm-85/qemu/target-i386/helper.c.vanilla	2009-05-05 17:36:13.000000000 +0200
+++ kvm-85/qemu/target-i386/helper.c	2009-05-05 17:36:37.000000000 +0200
@@ -29,6 +29,7 @@
 #include "exec-all.h"
 #include "qemu-common.h"
 #include "kvm.h"
+#include "hw/boards.h"
 
 #include "qemu-kvm.h"
 
@@ -1473,7 +1473,7 @@
          * isn't supported in compatibility mode on Intel.  so advertise the
          * actuall cpu, and say goodbye to migration between different vendors
          * is you use compatibility mode. */
-        if (kvm_enabled())
+        if (kvm_enabled() && current_machine != &mac_machine)
             host_cpuid(0, 0, NULL, ebx, ecx, edx);
         break;
     case 1:
